tests/ha/ctdb.pm

```diff
@@ -0,0 +1,122 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test ctdb resource agent
+# Maintainer: Julien Adamek <jadamek@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use hacluster;
+use version_utils 'is_sle';
+use utils 'systemctl';
+
+sub run {
+    my $cluster_name = get_cluster_name;
+    my $vip_ip       = '10.0.2.20';
+    my $ctdb_folder  = '/srv/fs_cluster_md/ctdb';
+
+    # CTDB configuration must be done only in cluster nodes
+    if (check_var('CTDB_TEST_ROLE', 'server')) {
+        my $ctdb_cfg = '/etc/samba/smb.conf';
+        my $ctdb_rsc = 'ctdb';
+        my $nmb_rsc  = 'nmb';
+        my $smb_rsc  = 'smb';
+        my $ip_rsc   = 'vip';
+        my @node_list;
+
+        foreach my $node (1 .. get_node_number) {
+            my $node_ip = get_ip(choose_node($node));
+            push @node_list, $node_ip;
+        }
+        my $node_list = 'echo -e "' . join("\n", @node_list) . '" > /etc/ctdb/nodes';
+        assert_script_run "$node_list";
```

Comment:
Did you check without `$node_list` variable with the `get_node_number` fix?

Reply:
No, I will give a try and I'll let you know.

Reply:
It works without `$node_list` variable, code updated thx.